WordPress/gutenberg
# Block Theme Previews: rename GET variable and prepare for core compat

## What?
When we backport the block theme previews to WordPress Core, we need to update the URL get parameters to `wp_theme_preview` instead o `gutenberg_theme_preview`. We should make the same change to Gutenberg so that we use the same param in both places. See https://github.com/WordPress/wordpress-develop/pull/4627.

## Why?
So that we have the same URL parameter in both places. 
Users that decide to stay on an older version of core after 6.3 but do update the GB plugin will lose access to the theme preview functionality unless we build support for both parameters. Since this hasn't been long in the plugin, I don't think there's much reason to maintain both, so the core version is preferred. Happy to hear any opinions against this decision.

## How?
Just rename the parameter.

## Testing Instructions
1. Open the theme showcase
2. Hover a block theme and confirm that you get a button allowing you preview the theme
3. Click the button and confirm that the site editor opens with the new theme being previewed
4. Check that when you save, the new theme is activated.
